relay-server/src/actors/outcome_aggregator.rs

```diff
@@ -187,6 +187,8 @@ impl Service for OutcomeAggregator {
 
             loop {
                 tokio::select! {
+                    // Prioritize flush over receiving messages to prevent starving. Shutdown can be
+                    // last since it is not vital if there are still messages in the channel.
```

Comment:
The documentation lists "shutdown" as an example of a message that should get priority, but I'm not sure if it applies in our case:

> If for example you are selecting between a stream and a shutdown future, and the stream has a huge volume of messages and zero or nearly zero time between them, you should place the shutdown future earlier in the select!

https://docs.rs/tokio/latest/tokio/macro.select.html#fairness

Reply:
In our case processing the shutdown message would not make sense if there are still messages pending because it would only lead to a premature flush.